sites/partners/__tests__/components/listings/PaperListingForm/UnitGroupForm.test.tsx

```diff
@@ -0,0 +1,329 @@
+    expect(screen.getByLabelText(/Min Square Footage/i)).toBeInTheDocument()
+    expect(screen.getByLabelText(/Max Square Footage/i)).toBeInTheDocument()
+    expect(screen.getByLabelText(/Minimum Floor/i)).toBeInTheDocument()
+    expect(screen.getByLabelText(/Maximum Floor/i)).toBeInTheDocument()
+    expect(screen.getByLabelText(/Min Number of Bathrooms/i)).toBeInTheDocument()
+    expect(screen.getByLabelText(/Max Number of Bathrooms/)).toBeInTheDocument()
+
+    // Availability Section
+    expect(screen.getByRole("heading", { level: 2, name: /availability/i })).toBeInTheDocument()
+
+    expect(screen.getByLabelText(/Unit Group Vacancies/i)).toBeInTheDocument()
+    expect(screen.getByText(/waitlist status/i)).toBeInTheDocument()
+    expect(screen.getByLabelText(/^open$/i)).toBeInTheDocument()
+    expect(screen.getByLabelText(/closed/i)).toBeInTheDocument()
+    expect(screen.getByRole("heading", { level: 2, name: /eligibility/i })).toBeInTheDocument()
+
+    // Eligibility Section
+    expect(screen.getByRole("heading", { level: 2, name: /eligibility/i })).toBeInTheDocument()
+    expect(screen.queryByRole("table")).not.toBeInTheDocument()
+    expect(screen.getByRole("button", { name: /add ami level/i }))
+  })
+
+  it("should open ami form drawer", async () => {
+    render(
+      <AuthProvider>
+        <FormProviderWrapper>
+          <UnitGroupForm
+            onClose={jest.fn()}
+            onSubmit={jest.fn()}
+            defaultUnitGroup={tempUnitGroup}
+            draft={true}
+            nextId={1}
+          />
+        </FormProviderWrapper>
+      </AuthProvider>
+    )
+
+    const addAmiButton = screen.getByRole("button", { name: /add ami level/i })
+    expect(addAmiButton).toBeInTheDocument()
+
+    await act(() => userEvent.click(addAmiButton))
+
+    const drawerHeader = screen.getByRole("heading", { level: 1, name: /add ami level/i })
+    expect(drawerHeader).toBeInTheDocument()
+
+    const drawerContainer = drawerHeader.parentElement.parentElement
+
+    expect(
+      within(drawerContainer).getByRole("heading", { level: 2, name: /^ami level$/i })
+    ).toBeInTheDocument()
+    expect(within(drawerContainer).getByLabelText(/ami chart/i)).toBeInTheDocument()
+    expect(within(drawerContainer).getByLabelText(/percentage of ami/i)).toBeInTheDocument()
+    expect(within(drawerContainer).getByText(/how is rent determined\?/i)).toBeInTheDocument()
+
+    const fixedAmountOption = within(drawerContainer).getByLabelText(/fixed amount/i)
+    const percentageOption = within(drawerContainer).getByLabelText(/% of income/i)
+    expect(fixedAmountOption).toBeInTheDocument()
+    expect(percentageOption).toBeInTheDocument()
+
+    await act(() => userEvent.click(fixedAmountOption))
+    expect(within(drawerContainer).getByLabelText(/monthly rent/i)).toBeInTheDocument()
+
+    await act(() => userEvent.click(percentageOption))
+    expect(within(drawerContainer).getAllByLabelText(/% of income/i)).toHaveLength(2)
```

Comment:
suggestion: Can you add another step here of clicking the "cancel" button to close the drawer?